src/core.c

```diff
@@ -4865,10 +4877,17 @@ static void WindowSizeCallback(GLFWwindow *window, int width, int height)
 
     if (IsWindowFullscreen()) return;
 
+    Vector2 windowScaleDPI = GetWindowScaleDPI();
+
     // Set current screen size
+#if defined(__APPLE__)
     CORE.Window.screen.width = width;
     CORE.Window.screen.height = height;
-    
+#else
+    CORE.Window.screen.width = width / windowScaleDPI.x;
+    CORE.Window.screen.height = height / windowScaleDPI.y;
```

Comment:
This callback gives the physical size (framebuffer size), so we need to divide by the dpi scale factor.